pkg/query/context.go

```diff
@@ -197,6 +245,75 @@ func (ctx *Context) wrapPathSeqForTracing(it Iterator, pathSeq PathSeq) PathSeq
 	}
 }
 
+// wrapPathSeqForAnalysis wraps a PathSeq to count results and track timing if analysis is enabled
+func (ctx *Context) wrapPathSeqForAnalysis(it Iterator, pathSeq PathSeq, opType string) PathSeq {
+	if ctx.Analyze == nil || it == nil {
+		return pathSeq
+	}
+
+	iterID := it.ID()
+
+	// Initialize stats for this iterator if not present
+	if _, exists := ctx.Analyze[iterID]; !exists {
+		ctx.Analyze[iterID] = AnalyzeStats{}
+	}
+
+	// Increment call counter based on operation type
+	stats := ctx.Analyze[iterID]
+	switch opType {
+	case "Check":
+		stats.CheckCalls++
+	case "IterSubjects":
+		stats.IterSubjectsCalls++
+	case "IterResources":
+		stats.IterResourcesCalls++
+	}
+	ctx.Analyze[iterID] = stats
+
+	// Wrap the PathSeq to count results and track timing
+	return func(yield func(Path, error) bool) {
+		startTime := time.Now()
+		resultCount := 0
+
+		for path, err := range pathSeq {
+			if err == nil {
+				resultCount++
+			}
+			if !yield(path, err) {
+				// Record partial results and timing before early exit
+				elapsed := time.Since(startTime)
+				ctx.recordAnalysisResults(iterID, opType, resultCount, elapsed)
+				return
+			}
+		}
+
+		// Record final result count and timing
+		elapsed := time.Since(startTime)
+		ctx.recordAnalysisResults(iterID, opType, resultCount, elapsed)
```

Comment:
```suggestion
		startTime := time.Now()
		resultCount := 0

		defer func() {
			elapsed := time.Since(startTime)
			ctx.recordAnalysisResults(iterID, opType, resultCount, elapsed)
		}()

		for path, err := range pathSeq {
			if err == nil {
				resultCount++
			}
			if !yield(path, err) {
				return
			}
		}
```